include/dr/concepts/concepts.hpp

```diff
@@ -36,11 +36,7 @@ concept distributed_iterator =
 
 template <typename R>
 concept remote_contiguous_range =
-    remote_range<R> && rng::random_access_range<R> &&
-    requires(R &r) {
-      lib::ranges::rank(r);
-      { lib::ranges::local(r) } -> rng::contiguous_range;
-    };
+    remote_range<R> && rng::random_access_range<R>;
```

Comment:
We'll talk about this tomorrow, but the important part of the distributed contiguous range concept is actually that there's a `local()` CPO that returns a contiguous range, so we don't want to remove that.  The key idea is that the local iterator might be a lot more efficient than the remote iterator.  (This is particularly true for multi-node, but may come up multi-GPU as well.)

We might consider adding a new concept `locally_accessible_range` or something that checks for `local`, since there might be non-contiguous ranges that have more efficient local iterators than remote.

Reply:
Then why does `distributed_range` not require `local()`?

What about `enumerate` or `zip` result? Shall we implement `zip_view` in such a way that the result of `enumerate`/`zip_view` supports `local` call?

We may discuss this on our next call if it is easier not to write long explanations here/

Reply:
I feel like `local()` should be optional, but not required, for `remote_range`.  We could add another concept called `locally_convertible_range` or something like that.  I haven't added it because it's not the highest priority at the moment.  (The primary advantage of `local()` is for the distributed memory/MPI case.)

`local()` has particular importance for shp at the moment because of a bug in our SYCL runtime that we can only get around by using raw pointers.